pyproject.toml

```diff
@@ -117,6 +108,7 @@ types-requests = "^2.28.11"
 types-simplejson = "^3.18.0"
 types-PyYAML = "^6.0.12"
 coverage = {extras = ["toml"], version = "^7.2"}
+pyarrow = {version = "^11.0.0", python = "<3.11"}
```

Comment:
PyArrow is available for Python 3.11, so why the restriction?

What are the implications of missing the PyArrow dependency? Presumably nothing significant since it was previously an optional dependency, but I'd like to know.

Reply:
Yeah, I think there used to be some conflicts between numpy and pyarrow but those seem resolved now. I'll remove the restriction.

Reply:
Oops, it's still the case: https://github.com/meltano/sdk/actions/runs/4359995790/jobs/7622418687

Reply:
> Do you know why we're using an old version of Numpy here?

@WillDaSilva Edited my comment above as you were typing 😅. It's because only such old version supports Python 3.7

Reply:
As a solution for the interim, we could add two dependencies for PyArrow: an old one for Python 3.7, and the one we have selected currently for all other versions of Python.